wandb/apis/public/runs.py

```diff
@@ -505,26 +585,34 @@ def load(self, force=False):
                 if self._attrs.get("summaryMetrics")
                 else {}
             )
-        except json.decoder.JSONDecodeError:
-            # ignore invalid utf-8 or control characters
-            self._attrs["summaryMetrics"] = json.loads(
-                self._attrs["summaryMetrics"],
-                strict=False,
+        except (json.decoder.JSONDecodeError, KeyError):
+            # Handle lightweight mode or invalid data gracefully
+            self._attrs["summaryMetrics"] = {}
+
+        try:
+            self._attrs["systemMetrics"] = (
+                json.loads(self._attrs["systemMetrics"])
+                if self._attrs.get("systemMetrics")
+                else {}
             )
-        self._attrs["systemMetrics"] = (
-            json.loads(self._attrs["systemMetrics"])
-            if self._attrs.get("systemMetrics")
-            else {}
-        )
+        except (json.decoder.JSONDecodeError, KeyError):
+            # Handle lightweight mode or invalid data gracefully
+            self._attrs["systemMetrics"] = {}
         if self._attrs.get("user"):
             self.user = public.User(self.client, self._attrs["user"])
+
         config_user, config_raw = {}, {}
-        for key, value in json.loads(self._attrs.get("config") or "{}").items():
-            config = config_raw if key in WANDB_INTERNAL_KEYS else config_user
-            if isinstance(value, dict) and "value" in value:
-                config[key] = value["value"]
-            else:
-                config[key] = value
+        try:
+            for key, value in json.loads(self._attrs.get("config") or "{}").items():
+                config = config_raw if key in WANDB_INTERNAL_KEYS else config_user
+                if isinstance(value, dict) and "value" in value:
+                    config[key] = value["value"]
+                else:
+                    config[key] = value
+        except (json.decoder.JSONDecodeError, KeyError):
```

Comment:
Don't use a large try-except like this for generic errors like `KeyError`. It should be scoped to just the code that might fail. I don't see how a `KeyError` is possible here.